workflow/controller/controller.go

```diff
@@ -1054,8 +1053,8 @@ func (wfc *WorkflowController) addWorkflowInformerHandlers(ctx context.Context)
 						log.WithError(err).Error("Failed to list pods")
 					}
 					for _, p := range podList.Items {
-						if err := wfc.enablePodForDeletion(ctx, pods, p.Namespace, p.Name, false); err != nil {
-							log.WithError(err).Error("Failed to enable pod for deletion")
+						if slice.ContainsString(p.Finalizers, common.FinalizerPodStatus) {
+							wfc.queuePodForCleanup(p.Namespace, p.Name, deletePod)
```

Comment:
hmm this is a change from the previous behavior to only queue instead of patching... but the `List` already incurred an API call...

this logic seems to stem from https://github.com/argoproj/argo-workflows/pull/12413#issuecomment-1897899247

I suppose this is fine so long as they are eventually deleted and the queue is more stable than multiple immediate network requests for every delete